app/src/main/java/com/czech/chronos/ui/components/SearchResultList.kt

```diff
@@ -10,67 +10,91 @@ import androidx.compose.material3.Checkbox
+import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.czech.chronos.network.models.CurrentTime
 import com.czech.chronos.ui.viewModels.SearchViewModel
+import com.czech.chronos.utils.DateUtil
 import com.czech.chronos.utils.Fonts
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+import java.time.ZoneId
+import java.time.ZonedDateTime
 
 @SuppressLint("StateFlowValueCalledInComposition")
 @RequiresApi(Build.VERSION_CODES.O)
 @Composable
 fun SearchResultList(
     list: List<CurrentTime>,
-    viewModel: SearchViewModel
+    onCheckedChange: (Boolean, CurrentTime) -> Unit,
 ) {
-    LazyColumn(
-        modifier = Modifier
-    ) {
+    LazyColumn {
         items(
             items = list,
         ) { data ->
-//            viewModel.updateTimeFromServer(data.timezoneLocation.toString())
-//
-//            val time by viewModel.timeFromTimeZoneState.collectAsState()
-//
-//            SearchResultItem(
-//                city = data.requestedLocation.toString(),
-//                cityTime = time,
-//                checked = data.checked,
-//                onCheckedChange = { checked ->
-//                    if (!checked) viewModel.deleteCurrentTimeFromDB(data.requestedLocation.toString())
-//                }
-//            )
+            var checked by remember {
+                mutableStateOf(data.checked)
+            }
+            SearchResultItem(
+                data = data,
+                checked = checked,
+                onCheckedChange = { newValue ->
+                    checked = newValue
+                    onCheckedChange(newValue, data)
+                }
+            )
         }
     }
 }
 
 @RequiresApi(Build.VERSION_CODES.O)
 @Composable
 fun SearchResultItem(
-    city: String,
-    cityTime: String,
+    data: CurrentTime,
     checked: Boolean,
     onCheckedChange: (Boolean) -> Unit,
+    modifier: Modifier = Modifier
 ) {
 
+    val timeFormatter = remember { DateUtil.timeFormat }
+
+    if (data.timezoneLocation.isNullOrEmpty()) {
+
+    }
+
+    var localTime by remember {
+        mutableStateOf(
+            ZonedDateTime.now(ZoneId.of(data.timezoneLocation))
+        )
+    }
+
+    LaunchedEffect(Unit) {
+        withContext(Dispatchers.IO) {
+            while (true) {
+                localTime = ZonedDateTime.now(ZoneId.of(data.timezoneLocation))
+                delay(1000L)
+            }
+        }
+    }
+
     Box(
-        modifier = Modifier
+        modifier = modifier
             .height(50.dp)
             .fillMaxWidth()
     ) {
         Row(
             horizontalArrangement = Arrangement.SpaceBetween,
             verticalAlignment = Alignment.CenterVertically,
-            modifier = Modifier
+            modifier = modifier
```

Comment:
Reusing `modifier` here can be a cause for subtle errors. Only use the `modifier` you receive as an argument on the outermost container.